connectors/source-http/internal/server.go

```diff
@@ -0,0 +1,205 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/google/uuid"
+	"net"
+	"net/http"
+	"sync"
+
+	v2 "github.com/cloudevents/sdk-go/v2"
+	cdkgo "github.com/linkall-labs/cdk-go"
+	"github.com/linkall-labs/cdk-go/log"
+	"github.com/valyala/fasthttp"
+)
+
+const (
+	name                       = "HTTP Source"
+	defaultPort                = 8080
+	reqSource                  = "source"
+	reqType                    = "type"
+	reqID                      = "id"
+	reqSubject                 = "subject"
+	reqSchema                  = "dataschema"
+	defaultSource              = "vanus-http-source"
+	defaultType                = "naive-http-request"
+	extendAttributesUserAgent  = "xvhttpuseragent"
+	extendAttributesRemoteIP   = "xvhttpremoteip"
+	extendAttributesRemoteAddr = "xvhttpremoteaddr"
+)
+
+var _ cdkgo.SourceConfigAccessor = &httpSourceConfig{}
+
+type httpSourceConfig struct {
+	cdkgo.SourceConfig `json:"_,inline" yaml:",inline"`
+	Port               int `json:"port" yaml:"port"`
+}
+
+func (c *httpSourceConfig) GetSecret() cdkgo.SecretAccessor {
+	return nil
+}
+
+func NewConfig() cdkgo.SourceConfigAccessor {
+	return &httpSourceConfig{}
+}
+
+var _ cdkgo.Source = &httpSource{}
+
+func NewHTTPSource() cdkgo.Source {
+	return &httpSource{
+		ch: make(chan *cdkgo.Tuple, 1024),
+	}
+}
+
+type httpSource struct {
+	cfg   *httpSourceConfig
+	mutex sync.Mutex
+	ch    chan *cdkgo.Tuple
+	ln    net.Listener
+}
+
+func (c *httpSource) Chan() <-chan *cdkgo.Tuple {
+	return c.ch
+}
+
+func (c *httpSource) Initialize(_ context.Context, cfg cdkgo.ConfigAccessor) error {
+	_cfg, ok := cfg.(*httpSourceConfig)
+	if !ok {
+		return errors.New("invalid config")
+	}
+
+	if _cfg.Port == 0 {
+		_cfg.Port = defaultPort
+	}
+	c.cfg = _cfg
+
+	ln, err := net.Listen("tcp", fmt.Sprintf(":%d", c.cfg.Port))
+	if err != nil {
+		return err
+	}
+	c.ln = ln
+	go func() {
+		log.Info("HTTP source is ready to serving", map[string]interface{}{
+			"listen": c.cfg.Port,
+		})
+		if err := fasthttp.Serve(ln, c.handleFastHTTP); err != nil {
+			log.Error("failed to start http server", map[string]interface{}{
+				log.KeyError: err,
+			})
+			panic(err)
+		}
+	}()
+	return nil
+}
+
+func (c *httpSource) Name() string {
+	return name
+}
+
+func (c *httpSource) Destroy() error {
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	if err := c.ln.Close(); err != nil {
+		log.Warning("close listener error", map[string]interface{}{
+			log.KeyError: err,
+		})
+	}
+	return nil
+}
+
+func (c *httpSource) handleFastHTTP(ctx *fasthttp.RequestCtx) {
+	path := string(ctx.Path())
+
+	if !ctx.IsPost() {
```

Comment:
why only support the HTTP method "post"?  we can put the HTTP method into the event attribute.

Reply:
I haven't seen any other scenarios to support other methods, do you have any idea?